main.sh

```diff
@@ -1,5 +1,24 @@
 #!/usr/bin/env bash
 
-export $(egrep -v '^#' .env | xargs)
+PORT=7092
+
+# https://stackoverflow.com/a/14203146
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    -p|--port)
+    PORT="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    *)    # unknown option
+    POSITIONAL+=("$1") # save it in an array for later
+    shift # past argument
+    ;;
+esac
```

Comment:
i dont think i see the benefit of all this. i personally believe that the deploy configuration should be the lowest common denominator.

i will have to try this in k8s. right now the deployment provides the port via an env variable.

Reply:
You should be able to provide `"command": ["--port", "XXXX"]` in the k8s yaml file if you need the container to use a particular port: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/, or my feeling is you should be able to forward the port wherever you need it at a higher level. But I'm just starting to play around with k8s.